tests/chainer_tests/functions_tests/loss_tests/test_vae.py

```diff
@@ -18,15 +22,19 @@ def setUp(self):
 
         # Refer to Appendix B in the original paper
         # Auto-Encoding Variational Bayes (https://arxiv.org/abs/1312.6114)
-        J = self.mean.size
-        self.expect = -(J + numpy.sum(self.ln_var) -
-                        numpy.sum(self.mean * self.mean) -
-                        numpy.sum(numpy.exp(self.ln_var))) * 0.5
+        loss = -(1 + self.ln_var -
+                 self.mean * self.mean -
+                 numpy.exp(self.ln_var)) * 0.5
+        if self.reduce == 'sum':
+            self.expect = numpy.sum(loss)
+        elif self.reduce == 'no':
+            self.expect = loss
 
     def check_gaussian_kl_divergence(self, mean, ln_var):
         m = chainer.Variable(mean)
         v = chainer.Variable(ln_var)
-        actual = cuda.to_cpu(vae.gaussian_kl_divergence(m, v).data)
+        actual = vae.gaussian_kl_divergence(m, v, self.reduce)
```

Comment:
As other functions in this file also have this problem, I fixed and sent it as another PR (#2541)